commands/host.vegman

```diff
@@ -193,35 +238,64 @@ function cmd_virtualmedia_mount {
     return 1
   fi
 
-  if [[ ! -e ${imagefile} ]] || [[ -d ${imagefile} ]]; then
-    echo "Please specify the path to image file" >&2
+  if ! [[ ${imagefile} =~ ${imagefile_pattern} ]]; then
+    echo "Please specify the valid path to the image file" >&2
     return 1
   fi
 
-  usb-ctrl insert "usb.ms.cli" "${imagefile}" "${type}"
+  if [[ ! -z ${username} ]] || [[ ! -z ${password} ]]
+  then
+    printf %b "${username:0:64}\0${password:0:64}\0" >6
+    cred_pipe=6
+  fi
+
+  $(busctl call xyz.openbmc_project.VirtualMedia \
+    /xyz/openbmc_project/VirtualMedia/Legacy/USB1 \
+    xyz.openbmc_project.VirtualMedia.Legacy \
+    Mount sbyv "$imagefile" $readonly ${type_id} y $cred_pipe  >/dev/null) && \
+  echo "The USB1 Virtual Media Device mounted"
 }
 
 # @sudo cmd_virtualmedia_umount admin
 # @restrict cmd_virtualmedia_umount admin
 # @doc cmd_virtualmedia_umount
 # Unmount the virtual media
+#   MEDIA     - the Virtual Media Device name to unmount (USB0 or USB1)
 #   -y, --yes - Do not ask for confirmation
 function cmd_virtualmedia_umount {
   local yes=""
-
+  local media=""
+  local dbus_vm_type=""
   while [[ $# -gt 0 ]]; do
     case "${1}" in
       -y | --yes) yes="y";;
-      *) abort_badarg "${1}";;
+      *)
+        if [[ -n "${media}" ]]; then
+          abort_badarg "${1}"
+        else
+          media="${1}"
+        fi
     esac
     shift
   done
+  if [[ "${media}" == "USB0" ]]
+  then
+    dbus_vm_type="Proxy"
+  elif [[ "${media}" == "USB1" ]]
+  then
+    dbus_vm_type="Legacy"
+  else
+    abort_badarg "${media}"
+  fi
 
   if [[ -z "${yes}" ]]; then
     confirm
   fi
 
-  usb-ctrl eject "usb.ms.cli"
+  busctl call xyz.openbmc_project.VirtualMedia \
+    "/xyz/openbmc_project/VirtualMedia/${dbus_vm_type}/${media}" \
+    "xyz.openbmc_project.VirtualMedia.${dbus_vm_type}" Unmount &>2 && \
+  echo "The ${media} Virtual Media Device unmounted"
```

Comment:
Here also `The` looks excess.
`Virtual Media Device ${media} unmounted`

Reply:
fixed